lib/src/signer.rs

```diff
@@ -0,0 +1,51 @@
+#[cfg(feature = "signing-gpgme")]
+use gpgme::{Context, Key, Protocol};
+
+use crate::backend::BackendResult;
+
+pub trait Signer {
+    fn sign(&self, input: &[u8]) -> BackendResult<String>;
```

Comment:
Should this really return a String? I would think it would return `Vec<u8>`.

Reply:
Yeah it should just return bytes, it's just that libgit was expecting a `&str`, i.e. valid utf, I should move converting to string up into the backend